PiwikTracker/Dispatcher.swift

```diff
@@ -0,0 +1,11 @@
+import Foundation
+
+internal protocol Dispatcher {
+    associatedtype Dispatchable
+    
+    var userAgent: String? { get set }
+    
+    func send(event: Dispatchable, success: ()->(), failure: (_ shouldContinue: Bool)->());
```

Comment:
Yes. In our case the Dispatchable is an event. This protocol just defines a Dispatcher as a thing that can dispatch one type (the Dispatchable) in a certain way. So there could be a NSURLSessionEventDispatcher, an AFNetworkingEventDispatcher or even a NSURLSessionSomethingCompletelyDifferentEventDispatcher. I guess in our case it doesn't matter since we only will dispatch Events, hm?